File: src/main/java/com/seaweed/identifier/user/UserService.java
package com.seaweed.identifier.user;

import com.seaweed.identifier.user.vo.UserVO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService{
    @Autowired
    private UserRepository userRepository;
    public List<UserVO> findUsers(Pageable pageable){
        return userRepository.findAll(pageable).getContent();
    }

    public long countUsers(){
        return userRepository.count();
    }

    public void saveUser(UserVO user){
        userRepository.save(user);
    }

    public void updateNameById(String id, String name){
        userRepository.updateNameById(id,name);
    }
}

File: src/main/kotlin/com/seaweed/identifier/user/UserService.kt
package com.seaweed.identifier.user

import org.springframework.stereotype.Service

@Service
class UserService(
    private val userRepository: UserRepository
) {

    fun findAllUsers() : List<User>{
        return userRepository.findAll();
    }

    fun saveUser(name : String, id : String){
        userRepository.save(User(name=name, id=id))
    }

    fun updateNameById(id:String, name: String){
        userRepository.updateNameById(id,name);
    }
}